packages/numbers/lib/numbers.ts

```diff
@@ -20,45 +25,93 @@ const runRequest = async <T>(options: NumbersClassParameters): Promise<NumbersRe
     return result;
 }
 
+const remapObjects = <T, O>(mapping, newObject: T, oldObject: O): T => {
+    for (let key in mapping) {
+        if (oldObject[mapping[key]]) {
+            newObject[key] = oldObject[mapping[key]];
+            delete oldObject[mapping[key]]
+        }
+    }
+    newObject = { ...newObject, ...oldObject };
+    return newObject;
+}
+
 const BASE_URL = "https://rest.nexmo.com".replace(/\/+$/, "");
 
 export const NumbersParamCreator = function (options?: NumbersClassParameters) {
     return {
         buyNumber(params?: NumbersParams) {
+            let mapping = { 'target_api_key': 'targetApiKey' };
+            let data: NumbersQueryParams = {
+                country: params.country,
+                msisdn: params.msisdn
+            };
+            data = remapObjects(mapping, data, params);
+
             const localVetchOptions = {};
             localVetchOptions['url'] = `${options.baseUrl}/number/buy`;
             localVetchOptions['headers'] = Object.assign({}, options.headers);
-            localVetchOptions['data'] = options.auth.getQueryParams(params);
+            localVetchOptions['data'] = options.auth.getQueryParams(data);
             localVetchOptions['method'] = 'POST';
             return localVetchOptions;
         },
         cancelNumber(params?: NumbersParams) {
+            let mapping = { 'target_api_key': 'targetApiKey' };
+            let data: NumbersQueryParams = {
+                country: params.country,
+                msisdn: params.msisdn
+            };
+            data = remapObjects(mapping, data, params);
```

Comment:
```suggestion
            data = remapObjects(mapping, data, params);
```
same issue as above of setting and passing same variable